.build-tools/cmd/check-lint-version_test.go

```diff
@@ -0,0 +1,88 @@
+package cmd
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseWorkflow(t *testing.T) {
+	t.Run("parse invalid workflow file", func(t *testing.T) {
+		parsedVersion, err := parseWorkflowVersionFromFile("../testdata/check-lint-version/invalid.yml")
+		assert.Equal(t, "", parsedVersion)
+		assert.Error(t, err)
+	})
+
+	t.Run("parse workflow file with a missing key", func(t *testing.T) {
+		parsedVersion, err := parseWorkflowVersionFromFile("../testdata/check-lint-version/invalid-test.yml")
+		assert.Equal(t, "", parsedVersion)
+		assert.NoError(t, err)
+	})
+
+	t.Run("parse an invalid workflow file", func(t *testing.T) {
+		parsedVersion, err := parseWorkflowVersionFromFile("../testdata/check-lint-version/invalid-yaml.yml")
+		assert.Equal(t, "", parsedVersion)
+		assert.Error(t, err)
+	})
+
+	t.Run("parse testing workflow file", func(t *testing.T) {
+		parsedVersion, err := parseWorkflowVersionFromFile("../../.github/workflows/test-tooling.yml")
+		assert.Equal(t, "v1.51.2", parsedVersion)
```

Comment:
I would remove this test because it is tightly coupled with the version in the YAML

Reply:
I've migrated the version to the testdata folder to still validate parsing a workflow file